src/main/java/org/opensearch/knn/index/codec/BasePerFieldKnnVectorsFormat.java

```diff
@@ -78,14 +81,10 @@ public KnnVectorsFormat getKnnVectorsFormatForField(final String field) {
             )
         ).fieldType(field);
 
-        KNNMappingConfig knnMappingConfig = mappedFieldType.getKnnMappingConfig();
-        KNNMethodContext knnMethodContext = knnMappingConfig.getKnnMethodContext()
-            .orElseThrow(() -> new IllegalArgumentException("KNN method context cannot be empty"));
-
-        final KNNEngine engine = knnMethodContext.getKnnEngine();
+        final KNNMethodContext knnMethodContext = extractKNNMethodContext(mappedFieldType);
         final Map<String, Object> params = knnMethodContext.getMethodComponentContext().getParameters();
```

Comment:
` final Map<String, Object> params = knnMethodContext.getMethodComponentContext().getParameters()`

getMethodComponentContext here can be null. We need to build this getMethodComponentContext from scratch. See BWC tests are failing.

Reply:
cc: @jmazanec15 , @shatejas